src/uint/div_limb.rs

```diff
@@ -0,0 +1,323 @@
+//! Implementation of constant-time division via reciprocal precomputation, as described in
+//! "Improved Division by Invariant Integers" by Niels Möller and Torbjorn Granlund
+//! (DOI: 10.1109/TC.2010.143, https://gmplib.org/~tege/division-paper.pdf).
+use subtle::{Choice, ConditionallySelectable, CtOption};
+
+use crate::{Limb, Uint, WideWord, Word};
+
+/// Returns `u16::MAX` if `x == y` and `0` otherwise.
+#[inline(always)]
+const fn ct_eq(x: u16, y: u16) -> u16 {
+    // e1 == 0 if and only if x == y
+    let e1 = x ^ y;
+
+    // If e1 == 0, then e1 and -e1 are both equal to zero;
+    // otherwise, one or both will have its high bit set.
+    let e2 = (e1 | e1.wrapping_neg()) >> (u16::BITS - 1);
+
+    // Result is the opposite of the high bit (now shifted to low).
+    (e2 ^ 1).wrapping_neg()
+}
+
+/// Returns `a` if `c == 0` and `b` if `c == u16::MAX`.
+#[inline(always)]
+const fn ct_select(a: u16, b: u16, c: u16) -> u16 {
+    a ^ (c & (a ^ b))
+}
+
+/// Constant-time element lookup in `arr`.
+#[inline(always)]
+const fn ct_lookup<const N: usize>(arr: &[u16; N], idx: u16) -> u16 {
+    let mut t = arr[0];
+    let mut j: u16 = 1;
+    while j < N as u16 {
+        t = ct_select(t, arr[j as usize], ct_eq(idx, j));
+        j += 1;
+    }
+    t
+}
```

Comment:
Curious if anyone has made a more efficient bitsliced implementation of those reciprocal tables

Reply:
That was a great suggestion, I reimplemented it via shift-and-subtract, and it's actually much faster this way.